### DISCUSSION TITLE
Health Check URL for Mirth Server in Docker

### DISCUSSION THREAD

**Discussion by rjcdoc** [nextgenhealthcare/connect#6275] [Q&A] [answered]
Is there any health check URL for a mirth server running in docker.  My idea was to just check the main URL where you launch the Mirth Administrator GUI.

**Comment by rjkroll** [comment, marked as answer]
You could try to hit the API URL, which in theory would only be accessible of Mirth was up and running normally:

`https://servername:8443/api`

You could also setup a HTTP listener channel in Mirth that returns a '200 OK Status' you could use as the URL Health-check.


EDIT:  I don't docker, but I forgot at one point I had setup a TCP Listener and used a Powershell script to send a TCP message to that endpoint, then scheduled it to run on a Windows server seperate from the Mirth server, and it sends a message every 1 minute.

**Reply by pacmano1** [reply]
Both work - but I usually stand up a dedicated listener for this function just so it is all viewable like any other channel.   You can get clever with this - i.e. as means to let a load balancer know that "this" server is fully up.  Requires more code of course.

**Reply by tranhan5010** [reply]
Thank you for the tip. For future reference, I made a simple canal for an internal healthcheck script. The port is configurable via `listener_healthcheck_port`, and the path is default to `/actuator/health`. Since no auth header is configured, the listener is limited to only localhost `127.0.0.1` interface.

[HealthcheckListener.xml](https://github.com/user-attachments/files/25442999/HealthcheckListener.xml)

**Reply by javirub** [reply]
> Thank you for the tip. For future reference, I made a simple canal for an internal healthcheck script. The port is configurable via `listener_healthcheck_port`, and the path is default to `/actuator/health`. Since no auth header is configured, the listener is limited to only localhost `127.0.0.1` interface.
> 
> [HealthcheckListener.xml](https://github.com/user-attachments/files/25442999/HealthcheckListener.xml)

What about using /mirth/api/server/status ? it returns a 200 and I think it's used just for that

**Reply by tranhan5010** [reply]
That would be better since it is native. However, I did not manage to find that API from documentation. Would you mind sharing it ?

Edit: I found it by clicking `View Client API` and looking into the OpenApi spec. Indeed there is an API `/server/status` with no auth. It would be better with some documentation about the value (what does `1` means. My test server returns `0` even though everything is up and running). In my case, we prefer a more uniform healthcheck API, so a custom channel is more appropriate.

<img width="1454" height="476" alt="image" src="https://github.com/user-attachments/assets/32ca682a-c3ca-4e17-a691-5f8190ef1b01" />
